packages/react-router-dom/index.tsx

```diff
@@ -1033,24 +1037,16 @@ function useScrollRestoration({
   // Save positions on unload
   useBeforeUnload(
     React.useCallback(() => {
-      if (state?.navigation.state === "idle") {
-        let key =
-          (getKey ? getKey(state.location, state.matches) : null) ||
-          state.location.key;
+      if (navigation.state === "idle") {
+        let key = (getKey ? getKey(location, matches) : null) || location.key;
```

Comment:
I realized when comparing the types that this was sending the internal `state.matches`, not `useMatches`

Reply:
moving to a new PR since it's functional changes